### DISCUSSION TITLE
Modular notification options not just Telegram

### DISCUSSION THREAD

**Discussion by f34rdotcom** [eusonlito/GPS-Tracker#69] [General]
Any interest in adding the ability to have multiple notification plugins/modules/drivers not just one?

I took a quick read of the code and this will not be a trivial update, but I am sure it would be a much used feature if added. 

I would be happy to help since I am using this to monitory my own E-Bike trips.

Best
 SM.

**Comment by eusonlito** [comment]
At the moment it's complicated for me to integrate a plugin system that supports notifications 😅.

What other notifications do you need? Maybe URL?

**Reply by f34rdotcom** [reply]
Ya not an easy feature to add for sure. Lots of places that are hard coded would need to be redesigned to be more dynamic so it will take some braining on a good path for sure, but still worth adding to the list :c) Easy solution is for me to modify the telegram client service code. 

I prefer Pushover for its being simple, easy, and cheap. I am sure others exist that all use a similar REST API. https://support.pushover.net/i44-example-code-and-pushover-libraries#php

I also like Twilio with SendGrid but the cost of time to set up for personal use is not very good. Once it is setup you can do crazy stuff like call multiple people ask questions via SMS or voice.
https://www.twilio.com/docs/libraries/reference/twilio-php/

Best
 SM
